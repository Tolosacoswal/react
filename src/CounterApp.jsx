import { useState} from 'react'
import { PropTypes } from "prop-types";



export const CounterApp = ({value}) => {
  console.log (`render`);
  const [ counter, setCounter,] = useState(value)

  const handleAdd = ()=> {
setCounter( (counter)=> counter + 2);
}
const handleSubstarct = () => setCounter(counter -2);
const handleReset = () => setCounter(value);



  
  
  return (
    <>
    <h1>CounterApp</h1>
    <h2>{ counter }</h2>
    
    <button onClick={ handleAdd }>+1</button>
    <button onClick={ handleSubstarct }>-1</button>
    <button onClick={ handleReset }>reset</button>

    




    </>
  )
}
CounterApp.prototype={
    CounterApp: PropTypes.string.isRequired, 
   value:PropTypes.string,
}
