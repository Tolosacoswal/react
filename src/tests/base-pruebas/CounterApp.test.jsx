
import { fireEvent, render, screen } from "@testing-library/react";

import { CounterApp } from "../../CounterApp";

describe('pruebas en <CounterApp/>', () => { 

        const initialValue = 10;
        test('debe de hacer match con el snapshot', () => { 
                const {container}= render(<CounterApp value = {initialValue}/>); expect (container).toMatchSnapshot();});

         
         test('debe mostrar el valor inicial de 100 <CounterApp value={100}>', () => { 
          render (<CounterApp value={100}/>);
          expect(screen.getByTestId(100)).toBeTruthy(); 
         });
         
         
         test('debe de incrementar con el botÃ³n +1', () => { 
                render( <CounterApp value = {initialValue}/>);
                fireEvent.click( screen.getAllByText(`+1`))
          });
          
 })
 