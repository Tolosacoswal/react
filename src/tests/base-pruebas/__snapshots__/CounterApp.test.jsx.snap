// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp/> debe hacer match con el snapshot 1`] = `undefined`;
